### PR TITLE
Update the dependencies to prepare for integration of emergency checkpointing

### PR DESCRIPTION
**Pull request by xuefgu**

### CODE REVIEW

**Review by gobbleturk — APPROVED** [review]
LGTM, please wait for approval from In-Ho as well

File: constraints_gpu.txt
Comment (chajath) [inline constraints_gpu.txt:133]: tensorflow version has been deliberately held back due to https://github.com/google/maxtext/issues/516

could you check if the issue has been resolved now? TL;DR OOMs on A3+ machines when using C4 dataset with tf-data

**Review by chajath — APPROVED** [review]

File: constraints_gpu.txt
Comment (xuefgu) [reply constraints_gpu.txt]: Withheld some package versions and there seems to be no issues. Waiting for the final check to pass before merging this PR.

**Pull request merged**
